docs/OV_Runtime_UG/supported_plugins/GPU.md

```diff
@@ -258,7 +275,47 @@ The code snippet below demonstrates how to use dynamic batching in simple scenar
+   - Unfusion of the fused subgraph when the fusing is illegal for runtime shape
+
+-  Higher memory consumption for in-memory cache
+
+   - Optimal kernels and weights from the previously used shapes are preserved in in-memory cache for future use
+
+
+Recommendations for performance improvement
+-----------------------------------------------------------
+
+- Use static shape whenever possible
+
+   - Static models can benefit from more agressive optimizations such as constant propagation, fusing, reorder optimization.
+   If the same shape is used for dynamic model, the performance is worse than running that shape with static model.
+   It is therefore recommended to reshape the dynamic model to a static model, if the scenario allows.
+
+- Use bounded dynamic shape whenever possible
+
+   - The GPU plugin needs to reallocate memory if the current shape is larger than the maximum of the previous shapes, which causes additional overhead.
+   - Using bounded dynamic shape will help reduce such overhead. For example, use {ov::Dimension(1, 10), ov::Dimension(1, 384)} instead of {ov::Dimension(-1), ov::Dimension(-1)}.
+   - Note that bounded dynamic *batch* is handled differently as mentioned above.
+
+- Use permanent cache, i.e., either OpenCL cache or OpenVino model_cache, to reduce runtime re-compilation overhead
+
+   - GPU plugin deploys in-memory cache to store compiled kernels for previously used shapes, but the size of such in-memory cache is limited.
+   Therefore, it is recommended to use either of the following permanent caches too:
+
+      - OpenVino model_cache: See the :doc:`Model caching overview<openvino_docs_OV_UG_Model_caching_overview>`. Note that this option is preferable for discrete GPUs, as it provides caching for OneDNN primitives not only for OpenCL primitives. 
+      - OpenCL cache : See the page : https://github.com/intel/compute-runtime/blob/master/opencl/doc/FAQ.md#feature-cl_cache
+
+- The longer the inference sequence, the better throughput can be obtained, because it can leverage the runtime compilation time.
+
+   - If the primitive has a shape-agnostic kernel and the static shape kernel for the current shape does not exist in the in-memory cache, the shape-agnostic kernel is used. Then, as mentioned above, optimal kernels for the current shapes are also asynchronously compiled in parallel for future use. If the application process removes the CompiledModel object and the GPU plugin is destroyed, any not-yet-started compilation tasks for optimal kernels will be canceled. However, if the application process allows enough time for the enqueued asynchronous compilation tasks, the more optimal kernels become available, enabling better throughput. For example, running 200 inputs of {[1, 1], ..., [1, 50], [1, 1], ... , [1, 50], [1, 1], ..., [1, 50], [1, 1], ..., [1, 50]} may achieve better throughput than running 100 inputs of {[1, 1], ..., [1, 50], [1, 1], ... , [1,50]}.
```

Comment:
I'm afraid this might be too much detail... we did not explain what `shape-agnostic` is...

Reply:
(shape-agnostic kernel is mentioned in the first section)

Reply:
oh I see ;) sorry for my ignorance.